src/Stormpath/Resource/Application.php

```diff
@@ -281,6 +286,74 @@ public function authenticate($username, $password, array $options = array())
     }
 
 
+    public function createIdSiteUrl(array $options = array())
+    {
+        if( ! isset( $options['callbackUri'] ) )
+            throw new InvalidCallbackUriException('Please provide a \'callbackUri\' in the $options array.');
+
+        $client = Client::getInstance();
```

Comment:
This won't work if you are using the ClientBuilder directly instead of the static configuration. So, if you need to use the dataStore, just add a "protected" method in the parent Resource class that returns the dataStore and get it from there. Something like this:

```php
protected function getDataStore() {
    return this->dataStore;
}
```

Then use that getter to get the RequestExecutor, etc.